### PR TITLE
[NPU] ernie-1.0 support npu device

### PR DESCRIPTION
**Pull request by pangyoki**
### PR types

Others

### PR changes

Others

### Description

为ernie-1.0模型添加NPU设备支持。

### CODE REVIEW

File: model_zoo/ernie-1.0/run_pretrain.py [hunk 1]
```diff
@@ -540,6 +546,7 @@ def do_train(args):
                         "softmax",
                         "layer_norm",
                         "gelu",
+                        "dropout",
```
Comment (ZHUI) [inline model_zoo/ernie-1.0/run_pretrain.py:552]: 额，这个 dropout 有必要吗？

File: model_zoo/ernie-1.0/run_pretrain.py [hunk 2]
```diff
@@ -600,7 +607,7 @@ def do_train(args):
             else:
                 optimizer.step()
 
-            optimizer.clear_grad()
+            optimizer.clear_grad(set_to_zero=False)
```
Comment (ZHUI) [inline model_zoo/ernie-1.0/run_pretrain.py]: why?

File: model_zoo/ernie-1.0/run_pretrain.py (hunk 1 above)
Comment (pangyoki) [reply model_zoo/ernie-1.0/run_pretrain.py:552]: 可以让dropout npu op中的Cast CANN op从AICPU算子变成AICore算子，提升性能。

File: model_zoo/ernie-1.0/run_pretrain.py (hunk 2 above)
Comment (pangyoki) [reply model_zoo/ernie-1.0/run_pretrain.py]: 在NPU上执行时，设置set_to_zero=False可以提升性能。加了npu设备的判断条件。

**Review by ZHUI — APPROVED** [review]
LGTM
